[BUG]:

### Tracer Version(s)

5.69.0

### Node.js Version(s)

24.7.0

### Bug Report

dd-trace 5.67.0 -> 5.69.0 breaks with Playwright 1.56.0
R

### Reproduction Code

_No response_

### Error Logs

un npx playwright test --project 'regression-homepage' --config=playwrightConfigs/regression.playwright.config.ts --shard=1/6
TypeError: rootSuite.allTests is not a function
    at newCreateRootSuite (/home/runner/work/di-playwright-automation/di-playwright-automation/node_modules/dd-trace/packages/datadog-instrumentations/src/playwright.js:777:32)
    at /home/runner/work/di-playwright-automation/di-playwright-automation/node_modules/dd-trace/packages/datadog-instrumentations/src/playwright.js:621:29

playwright-ctrf-json-reporter: successfully written ctrf json to ../ctrf_reports/mazda/ctrf-report.json
Error: Process completed with exit code 1.

### Tracer Config

_No response_

### Operating System

_No response_

### Bundling

Unsure

--- a/packages/datadog-instrumentations/src/playwright.js
+++ b/packages/datadog-instrumentations/src/playwright.js
@@ -772,7 +772,10 @@ addHook({
     if (!isKnownTestsEnabled && !isTestManagementTestsEnabled && !isImpactedTestsEnabled) {
       return oldCreateRootSuite.apply(this, arguments)
     }
-    const rootSuite = await oldCreateRootSuite.apply(this, arguments)
+
+    const createRootSuiteReturnValue = await oldCreateRootSuite.apply(this, arguments)
+    // From v1.56.0 on, createRootSuite returns `{ rootSuite, topLevelProjects }`
+    const rootSuite = createRootSuiteReturnValue.rootSuite || createRootSuiteReturnValue
 
     const allTests = rootSuite.allTests()
 
@@ -861,7 +864,7 @@ addHook({
       }
     }
 
-    return rootSuite
+    return createRootSuiteReturnValue
   }
 
   // We need to proxy the createRootSuite function because the function is not configurable

